js_modules/dagit/packages/core/src/workspace/WorkspaceAssetsRoot.tsx

```diff
@@ -0,0 +1,132 @@
+import {gql, useQuery} from '@apollo/client';
+import {Box, Colors, NonIdealState, Spinner, TextInput} from '@dagster-io/ui';
+import * as React from 'react';
+
+import {PYTHON_ERROR_FRAGMENT} from '../app/PythonErrorInfo';
+import {useTrackPageView} from '../app/analytics';
+
+import {VirtualizedAssetTable} from './VirtualizedAssetTable';
+import {WorkspaceHeader} from './WorkspaceHeader';
+import {repoAddressToSelector} from './repoAddressToSelector';
+import {RepoAddress} from './types';
+import {WorkspaceAssetsQuery, WorkspaceAssetsQueryVariables} from './types/WorkspaceAssetsQuery';
+
+export const WorkspaceAssetsRoot = ({repoAddress}: {repoAddress: RepoAddress}) => {
+  useTrackPageView();
+
+  const [searchValue, setSearchValue] = React.useState('');
+  const selector = repoAddressToSelector(repoAddress);
+
+  const queryResultOverview = useQuery<WorkspaceAssetsQuery, WorkspaceAssetsQueryVariables>(
+    WORKSPACE_ASSETS_QUERY,
+    {
+      fetchPolicy: 'network-only',
+      notifyOnNetworkStatusChange: true,
+      variables: {selector},
+    },
+  );
+  const {data, loading} = queryResultOverview;
+
+  const sanitizedSearch = searchValue.trim().toLocaleLowerCase();
+  const anySearch = sanitizedSearch.length > 0;
+
+  const assetNodes = React.useMemo(() => {
+    if (data?.repositoryOrError.__typename === 'Repository') {
+      return data.repositoryOrError.assetNodes;
+    }
+    return [];
+  }, [data]);
+
+  const filteredBySearch = React.useMemo(() => {
+    const searchToLower = sanitizedSearch.toLocaleLowerCase();
+    return assetNodes.filter(({assetKey}) =>
+      assetKey.path.some((elem) => elem.toLocaleLowerCase().includes(searchToLower)),
```

Comment:
Hmm, we don't have to change this now but I think this should match AssetsCatalogTable.tsx:117:

```
  const searchPath = (searchValue || '')
    .replace(/(( ?> ?)|\.|\/)/g, '/')
    .toLowerCase()
    .trim();


        const pathMatch = !searchPath || tokenForAssetKey(a.key).toLowerCase().includes(searchPath);
```

It'd actually be nice to move this "does this asset key match this search string" out to a helper method because it's been through a bunch of iterations to make it more permissive / less permissive and I think we landed with that somehow. (or at least, if we change it again it'd be nice for it to change on both pages)